server/manifestmergetool/src/test/java/com/defold/manifestmergetool/ManifestMergeToolTest.java

```diff
@@ -169,7 +169,7 @@ public void testMergeAndroid() throws IOException {
                 + "          android:configChanges=\"keyboard|keyboardHidden|screenLayout|screenSize|orientation\""
                 + "          android:label=\"Test Project\" />"
                 + "    </application>"
-                + "    <uses-permission android:name=\"android.permission.CAMERA\" tools:node=\"remove\"/>"
+                + "    <uses-permission android:name=\"android.permission.CAMERA\"/>"
```

Comment:
Started from manifest merger 30.+ markers defined in library manifest doesn't affects higher priority manifests (e.g. main manifest). So all merge markers should be placed in main manifest. Libraries markers affects only manifests that have lower priority.
I'm not sure is it a breaking change. From my point of view it's correct behavior to define markers in main (application manifest) rather than in library manifest because library doesn't know anything about other libraries.

Reply:
Hmm, I'm not sure what implications this might have for extensions. @AGulev ?

Reply:
It's a good question. I don't remember if we use this anywhere. let me check

Reply:
I see one in firebase 
https://github.com/defold/extension-firebase/blob/ad44dde1c6be98824a19266fbb6191aaf15df9a2/firebase/manifests/android/AndroidManifest.xml#L11

not sure how it suppose to work in this case then?

Reply:
I can't find information about such changes in manifest merger tool 30+, could you share a link pls ?

Reply:
I can't find any changelogs for it. I compare how different version works. Started from 30.+ `tool:node="remove"` in library manifest stop affects result manifest. Only if that definition moved into application manifest - it affects merge result.

Reply:
I've checked how it works in firebase case. I've built extension example project as-is.
When I added `firebase.no_auto_init` - provider was removed as expected. Without that section - provider is presented.

One more thing that I noticed (doesn't relate to current discussing): doesn't matter if `firebase.no_auto_init = 1` or `firebase.no_auto_init = 0`. If section is presented and not empty - provider removed. I think it happens because Bob doesn't convert 1 or 0 to boolean values when render template

Reply:
Manifest merge command looks like 
`
java -jar /etc/extender/apps/manifestmergetool.jar --platform android --main /tmp/job3918344856207810552/upload/AndroidManifest.xml  --lib /tmp/job3918344856207810552/upload/firebase/manifests/android/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.firebase-firebase-installations-17.2.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.lifecycle-lifecycle-livedata-2.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.firebase-firebase-installations-interop-17.1.1.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.loader-loader-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.arch.core-core-runtime-2.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.legacy-legacy-support-core-ui-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.lifecycle-lifecycle-runtime-2.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.slidingpanelayout-slidingpanelayout-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.firebase-firebase-common-ktx-20.4.2.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.firebase-firebase-common-20.4.2.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.interpolator-interpolator-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.customview-customview-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.viewpager-viewpager-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.legacy-legacy-support-core-utils-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.asynclayoutinflater-asynclayoutinflater-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.localbroadcastmanager-localbroadcastmanager-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.fragment-fragment-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.cursoradapter-cursoradapter-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.core-core-1.2.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.lifecycle-lifecycle-viewmodel-2.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.drawerlayout-drawerlayout-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.android.gms-play-services-base-18.2.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.firebase-firebase-components-17.1.5.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.documentfile-documentfile-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.android.gms-play-services-tasks-18.0.2.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.swiperefreshlayout-swiperefreshlayout-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.versionedparcelable-versionedparcelable-1.1.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.print-print-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/com.google.android.gms-play-services-basement-18.1.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.coordinatorlayout-coordinatorlayout-1.0.0.aar/AndroidManifest.xml  --lib /tmp/.gradle/unpacked/androidx.lifecycle-lifecycle-livedata-core-2.0.0.aar/AndroidManifest.xml  --out /tmp/job3918344856207810552/build/AndroidManifest.xml
`

Reply:
ah, I think I understand why it works in case with firebase:
we merge extension's manifest into the main manifest, so now it's higher prio that the rest manifests it will be merged with...